packages/opentelemetry-basic-tracer/src/Span.ts

```diff
@@ -132,4 +156,48 @@ export class Span implements types.Span, ReadableSpan {
     }
     return this._ended;
   }
+
+  // Converts a number to HrTime
+  private static _numberToHrtime(time: number): types.HrTime {
+    const millis = Math.trunc(time)
+    const nanos = Number((time - millis).toFixed(2)) * 1e+10;
+    return [millis, nanos];
+  }
+
+  // Converts a TimeInput to an HrTime, defaults to _hrtime().
+  private static _toHrTime(time?: types.TimeInput): types.HrTime {
+    if (Array.isArray(time)) {
+      return time;
+    } else if (typeof time === 'number') {
+      // Must be a performance.now() if it's smaller than process start time.
+      if (time < performance.timeOrigin) {
+        return Span._hrtime(time);
+      }
+      // epoch milliseconds or performance.timeOrigin
+      else {
+        return Span._numberToHrtime(time);
+      }
+    } else if (time instanceof Date) {
+      return [time.getTime(), 0];
+    } else {
+      return Span._hrtime();
+    }
+  }
+
+  // Returns an hrtime calculated via performance component.
+  private static _hrtime(performanceNow?: number): types.HrTime {
+    const timeOrigin = Span._numberToHrtime(performance.timeOrigin);
+    const now = Span._numberToHrtime(performanceNow || performance.now());
+
+    let millis = timeOrigin[0] + now[0];
+    let nanos = timeOrigin[1] + now[1];
+
+    // Nanoseconds
+    if (nanos > 1e+10) {
```

Comment:
I'm slightly confused, about `performance.now()`. Is it sub-nanosecond accurate or nanoseconds?
Micro to nano is 1e+6. If it's subnano it's not compatible with `process.hrtime` and we need to multiple.

`node -p -e "require('perf_hooks').performance.now()"` gives me `62.502530097961426` which is 15 digits aftr the decimal, and Chrome's `performance.now()` gives me `261435.1250003092`. which is 10 digits. Should we round to nanoseconds (1e+6) andcleave sub-nanos?

Reply:
I would guess that `performance.now` is nanosecond accuracy under the hood but due to rounding issues, it has longer decimal representations (I could be wrong on this).

Anyway, I think nanosecond accuracy should be sufficient for tracing even really short or time sensitive events.

I do think rounding here is appropriate once we make it into a tuple.